openff/interchange/interop/internal/amber.py

```diff
@@ -0,0 +1,630 @@
+"""Interfaces with Amber."""
+import textwrap
+from pathlib import Path
+from typing import TYPE_CHECKING, Dict, Union
+
+import numpy as np
+from openff.units import unit
+
+from openff.interchange.components.mdtraj import _get_num_h_bonds, _store_bond_partners
+
+if TYPE_CHECKING:
+    from openff.interchange.components.interchange import Interchange
+    from openff.interchange.models import PotentialKey
+
+
+AMBER_COULOMBS_CONSTANT = 18.2223
+kcal_mol = unit.kilocalorie / unit.mol
+kcal_mol_a2 = kcal_mol / unit.angstrom ** 2
+kcal_mol_rad2 = kcal_mol / unit.radian ** 2
+
+
+def _write_text_blob(file, blob):
+    if blob == "":
+        file.write("\n")
+    else:
+        for line in textwrap.wrap(blob, width=80, drop_whitespace=False):
+            file.write(line + "\n")
+
+
+def _get_exclusion_lists(topology):
+
+    _store_bond_partners(topology.mdtop)
+
+    number_excluded_atoms = list()
+    excluded_atoms_list = list()
+
+    for atom1 in topology.mdtop.atoms:
+        # Excluded atoms _on this atom_
+        tmp = list()
+
+        for atom2 in atom1._bond_partners:
+
+            if atom2.index > atom1.index and atom2.index + 1 not in tmp:
+                tmp.append(atom2.index + 1)
+
+            for atom3 in atom2._bond_partners:
+
+                if atom3.index > atom1.index and atom3.index + 1 not in tmp:
+                    tmp.append(atom3.index + 1)
+
+                for atom4 in atom3._bond_partners:
+                    if atom4.index > atom1.index and atom4.index + 1 not in tmp:
+                        tmp.append(atom4.index + 1)
+
+        if len(tmp) == 0:
+            tmp.append(0)
+
+        number_excluded_atoms.append(len(tmp))
+        [excluded_atoms_list.append(_) for _ in tmp]
+
+    return number_excluded_atoms, excluded_atoms_list
+
+
+def to_prmtop(interchange: "Interchange", file_path: Union[Path, str]):
```

Comment:
You might want to consider splitting this mono-function into smaller focused ones to enable precise unit testing!

Reply:
Definitely 👍